include/openthread/thread.h

```diff
@@ -1114,6 +1122,33 @@ void otThreadSetStoreFrameCounterAhead(otInstance *aInstance, uint32_t aStoreFra
  */
 uint32_t otThreadGetStoreFrameCounterAhead(otInstance *aInstance);
 
+/**
+ * Attempts to wake a Wake-up End Device.
+ *
+ * Requires `OPENTHREAD_CONFIG_WAKEUP_COORDINATOR_ENABLE` to be enabled.
+ *
+ * The wake-up starts with transmitting a wake-up frame sequence to the Wake-up End Device.
+ * During the wake-up sequence, and for a short time after the last wake-up frame is sent, the Wake-up Coordinator keeps
+ * its receiver on to be able to receive an initial mesh link establishment message from the WED.
+ *
+ * @param[in] aInstance         A pointer to an OpenThread instance.
+ * @param[in] aWedAddress       The extended address of the Wake-up End Device.
+ * @param[in] aWakeupIntervalUs An interval between consecutive wake-up frames (in microseconds).
+ * @param[in] aWakeupDurationMs Duration of the wake-up sequence (in milliseconds).
+ * @param[in] aCallback         A pointer to function that is called when the wake-up succeeds or fails.
+ * @param[in] aContext          A pointer to callback application-specific context.
+ *
+ * @retval OT_ERROR_NONE          Successfully started the wake-up.
+ * @retval OT_ERROR_INVALID_STATE Another attachment request is still in progress.
+ * @retval OT_ERROR_INVALID_ARGS  The wake-up interval or duration are invalid.
```

Comment:
Does this function support setting `aWedAddress = null`?

Reply:
It doesn't but it also doesn't error out in such a case - like for all core types in OT, there's an assertion when trying to cast a null pointer to a reference.

Reply:
That's should be fine not to check the `aWedAddress = null`.